pkg/gameservers/controller.go

```diff
@@ -474,6 +477,35 @@ func (c *Controller) syncGameServerPortAllocationState(ctx context.Context, gs *
 	return gs, nil
 }
 
+// Allocate gameserver ports ( containerports ) for dynamic protocols such as TCPUDP
+// TODO: Going forward, this needs to be enhanced for other kinds of dynamic protocols
+func handleTCPUDPPortAllocationforStaticPortPolicy(gs *agonesv1.GameServer) {
```

Comment:
I thought we were going to move this logic into `GameServer.Pod(...)`?

https://github.com/googleforgames/agones/issues/2314#issuecomment-1055714041

Reply:
The logic for TCPUDP port allocation for static port policy is encapsulated in the method “**handleTCPUDPPortAllocationforStaticPortPolicy”**. This method is invoked from **“syncGameServerPortAllocationState”**. Since this is an explicit port allocation action this needs to happen before the event “port  allocated” is recorded, **_“ c.recorder.Event(gs, corev1.EventTypeNormal, string(gs.Status.State), "Port allocated") “_**

Reply:
Got it. That makes a certain amount of sense.

I'm not 100% sure, but is there a way to move this duplication of port logic into some kind of centralised function(s) that are shared between the two implementations? Feels like a lot of duplication.

But I can't see it myself, but I thought I'd ask the question?